editor/components/rich-text/format-toolbar/index.js

```diff
@@ -131,28 +130,33 @@ class FormatToolbar extends Component {
 	}
 
 	addLink() {
-		this.setState( { isEditingLink: false, isAddingLink: true, newLinkValue: '' } );
+		this.setState( { newLinkValue: '' } );
+		this.props.onChange( { link: { isAdding: true } } );
 	}
 
 	dropLink() {
 		this.props.onChange( { link: undefined } );
-		this.setState( { isEditingLink: false, isAddingLink: false, newLinkValue: '' } );
+		this.setState( { newLinkValue: '' } );
```

Comment:
What is `newLinkValue` used for? Not related to this PR, but this isn't really clear to me.